applications/StructuralMechanicsApplication/custom_elements/membrane_element.cpp

```diff
@@ -0,0 +1,1271 @@
+            const double integration_weight_i = r_integration_points[point_number].Weight();
+            const Matrix& shape_functions_gradients_i = r_shape_functions_gradients[point_number];
+            CovariantBaseVectors(base_vectors_current_cov,shape_functions_gradients_i,ConfigurationType::Reference);
+            base_1 += base_vectors_current_cov[0]*integration_weight_i;
+            base_2 += base_vectors_current_cov[1]*integration_weight_i;
+        }
+
+        MathUtils<double>::CrossProduct(base_3, base_1, base_2);
+        base_3 /= MathUtils<double>::Norm(base_3);
+
+        column(rOutput,0) = base_1;
+        column(rOutput,1) = base_2;
+        column(rOutput,2) = base_3;
+    }
+}
+
+void MembraneElement::GetValueOnIntegrationPoints(
+    const Variable<array_1d<double, 3>>& rVariable,
+    std::vector<array_1d<double, 3>>& rOutput,
+    const ProcessInfo& rCurrentProcessInfo)
+{
+    KRATOS_TRY;
+    CalculateOnIntegrationPoints(rVariable, rOutput, rCurrentProcessInfo);
+    KRATOS_CATCH("")
+}
+
+void MembraneElement::GetValueOnIntegrationPoints(const Variable<Vector>& rVariable,
+                        std::vector<Vector>& rValues,
+                        const ProcessInfo& rCurrentProcessInfo)
+{
+    KRATOS_TRY;
+    CalculateOnIntegrationPoints(rVariable, rValues, rCurrentProcessInfo);
+    KRATOS_CATCH("")
+}
+
+void MembraneElement::CalculateMassMatrix(MatrixType& rMassMatrix, ProcessInfo& rCurrentProcessInfo)
+{
+    KRATOS_TRY
+    auto& r_geom = GetGeometry();
+
+    // LUMPED MASS MATRIX
+    SizeType number_of_nodes = r_geom.size();
+    SizeType mat_size = number_of_nodes * 3;
+
+    if (rMassMatrix.size1() != mat_size) {
+        rMassMatrix.resize(mat_size, mat_size, false);
+    }
+
+    noalias(rMassMatrix) = ZeroMatrix(mat_size, mat_size);
+
+    const double total_mass = r_geom.Area() * GetProperties()[THICKNESS] *
+        StructuralMechanicsElementUtilities::GetDensityForMassMatrixComputation(*this);
+
+    Vector lump_fact =  ZeroVector(number_of_nodes);
+    r_geom.LumpingFactors(lump_fact);
+
+    for (SizeType i = 0; i < number_of_nodes; ++i) {
+        const double temp = lump_fact[i] * total_mass;
+
+        for (SizeType j = 0; j < 3; ++j)
+        {
+            const SizeType index = i * 3 + j;
+            rMassMatrix(index, index) = temp;
+        }
+    }
+
+    KRATOS_CATCH("")
+}
+
+void MembraneElement::CalculateLumpedMassVector(VectorType& rMassVector)
+{
+    KRATOS_TRY
+    auto& r_geom = GetGeometry();
+    const SizeType dimension = r_geom.WorkingSpaceDimension();
+    const SizeType number_of_nodes = r_geom.size();
+    const SizeType local_size = dimension*number_of_nodes;
+
+    if (rMassVector.size() != local_size) {
+        rMassVector.resize(local_size, false);
+    }
+
+    const double total_mass = r_geom.Area() * GetProperties()[THICKNESS] * StructuralMechanicsElementUtilities::GetDensityForMassMatrixComputation(*this);;
```

Comment:
just to add that this is technically wrong, since you don't get the initial but the current area

Reply:
yes but this is a problem in all structmech elements. We should really do sth. about that